edgedb/_testbase.py

```diff
@@ -52,54 +54,74 @@ def flt(log_record):
 _default_cluster = None
 
 
-def _init_cluster(data_dir=None, *, cleanup_atexit=True):
-    if (not os.environ.get('EDGEDB_DEBUG_SERVER') and
-            not os.environ.get('EDGEDB_LOG_LEVEL')):
-        _env = {'EDGEDB_LOG_LEVEL': 'silent'}
-    else:
-        _env = {}
+def _start_cluster(*, cleanup_atexit=True):
+    global _default_cluster
 
-    if data_dir is None:
-        cluster = edgedb_cluster.TempCluster(env=_env, testmode=True)
-        destroy = True
-    else:
-        cluster = edgedb_cluster.Cluster(data_dir=data_dir, env=_env)
-        destroy = False
+    if isinstance(_default_cluster, Exception):
+        # when starting a server fails
+        # don't retry starting one for every TestCase
+        # because that takes a long time
+        raise _default_cluster
 
-    if cluster.get_status() == 'not-initialized':
-        cluster.init()
+    if _default_cluster:
+        return _default_cluster
 
-    cluster.start(port='dynamic')
-    cluster.set_superuser_password('test')
+    try:
+        status_file = pathlib.Path('/tmp/edgedb-server-status')
+        if sys.platform == 'win32':
+            status_file = pathlib.Path('/mnt/c/temp/edgedb-server-status.txt')
+
+        args = [
+            os.environ.get('EDGEDB_SERVER_BINARY', 'edgedb-server'),
+            "--temp-dir",
+            "--testmode",
+            f"--emit-server-status={status_file}",
+            "--port=auto",
+            "--bootstrap-command=ALTER ROLE edgedb { SET password := 'test' }",
+        ]
+        cwd = '/tmp'
+
+        if sys.platform == 'win32':
+            args = ['wsl', 'sudo', '-u', 'edgedb'] + args
```

Comment:
`wsl.exe` supports the `-u` flag directly, you don't need to do `sudo`, I think.

Reply:
changed